include/crpropa/module/Observer.h

```diff
@@ -204,6 +204,19 @@ class ObserverElectronVeto: public ObserverFeature {
 	std::string getDescription() const;
 };
 
+/**
+ @class ObserverCustomVeto
+ @brief Custom veto for user-defined particle types.
+ */
+class ObserverCustomVeto: public ObserverFeature {
+private:
+	int vetoParticleId;
+public:
+	ObserverCustomVeto(int pId);
```

Comment:
Sorry for missing this one before. Can you please rename it to something more specific, to include `ParticleId` so that it is consistent with other modules? For example, `ObserverParticleIdVeto` or something similar...

Reply:
Good idea. Done with commit [298bf52](https://github.com/CRPropa/CRPropa3/pull/289/commits/298bf5291fc6457a2cb1a10b22a6c7e8a2043d12)